pkg/kube/snapshot/snapshot_alpha.go

```diff
@@ -417,19 +417,27 @@ func TransformUnstructured(u *unstructured.Unstructured, value interface{}) erro
 	return errors.Wrapf(err, "Failed to Unmarshal unstructured object")
 }
 
-func getSnapshotClassbyAnnotation(dynCli dynamic.Interface, kubeCli kubernetes.Interface, gvr schema.GroupVersionResource, annotationKey, annotationValue, storageClass string) (string, error) {
+func GetSnapshotClassbyAnnotation(dynCli dynamic.Interface, kubeCli kubernetes.Interface, gvr schema.GroupVersionResource, annotationKey, annotationValue, storageClass string) (string, error) {
+	// fetch storageClass
+	sc, err := kubeCli.StorageV1().StorageClasses().Get(context.TODO(), storageClass, metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Errorf("Failed to find StorageClass (%s) in the cluster: %v", storageClass, err)
+	}
+	// Check if storageclass annotation override is present.
+	if val, ok := sc.Annotations[annotationKey]; ok {
+		vsc, err := dynCli.Resource(gvr).Get(context.TODO(), val, metav1.GetOptions{})
+		if err != nil {
```

Comment:
Should we check the annotation value and the driver ( we check for this later at Line 447 ) ? 

```
if val, ok := ans[annotationKey]; ok && val == annotationValue && driver == sc.Provisioner {
```

Reply:
No this may not always be the case (even though it mostly is). The below check is more strict and enforces that. This is an override that allows any volumesnapshotclass to be specified

Reply:
Sounds good.